assets/js/src/marketing_optin_block/block.json

```diff
@@ -0,0 +1,27 @@
+{
+  "apiVersion": 2,
+  "name": "mailpoet/marketing-optin-block",
+  "version": "0.1.0",
+  "title": "MailPoet Marketing Opt-in",
+  "category": "woocommerce",
+  "textdomain": "mailpoet",
+  "supports": {
+    "align": false,
+    "html": false,
+    "multiple": false,
+    "reusable": false,
+    "inserter": false
+  },
+  "attributes": {
+    "lock": {
+      "type": "object",
+      "default": {
+        "remove": true,
+        "move": false
+      }
+    }
+  },
+  "parent": ["woocommerce/checkout-contact-information-block"],
+  "editorScript": "file:../../../dist/js/marketing_optin_block/marketing-optin-block.js",
+  "editorStyle": "file:../../../dist/js/marketing_optin_block/marketing-optin-block.css"
```

Comment:
Could you please elaborate on how this works? Where would this file go? All of our built files are in the `dist` folder, would this path still be valid?

Also, for all of our built assets, we use a Webpack plugin that adds a random string into the file name. This prevents a caching problems when releasing a new version. Is this a concern here as well?

Reply:
this is file is consumed by WordPress native function `register_block_type`, it would use this path to load the correct files.
You can see that the file reference `dist`.

> Also, for all of our built assets, we use a Webpack plugin that adds a random string into the file name. This prevents a caching problems when releasing a new version. Is this a concern here as well?

We were not aware of that. It would be lovely to implement yet we're a bit constrained to actually do this. The above files have a `?ver=version` attached to them, as in the standard in WordPress, this should work just fine. Hashing in file names is superior but would the above block a bit complicated, I believe this can be a follow up?